content/essentials/intro-to-reactivity.mdx

```diff
@@ -0,0 +1,171 @@
+
+## Synchronous vs. asynchronous
+
+Reactivity in systems can manifest primarily in two ways: synchronous and asynchronous. 
+The choice between these often depend on the requirements of the system and the nature of the tasks involved.
+
+### Synchronous reactivity
+Synchronous reactivity operates in a direct and linear fashion. 
+When a signal changes, its corresponding effects are immediately updated in a sequential manner. 
+This form of reactivity provides a direct correlation between a signal's change and the system's response, guaranteeing that the system remains consistently aligned with the most recent data.
+
+### Asynchronous reactivity
+Asynchronous reactivity introduces a layer of flexibility and adaptability. 
+Unlike it synchronous counterpart, the updates to the effects are not instantaneous. 
+Instead, they may be delayed or could wait for the completion of an external event or task.
+
+This approach is useful in scenarios where an effect depends on multiple signals and when updating one signal before another may result a data inconsistency. 
+In such a scenario, [`batch`](/reference/batch) can be used to delay triggering effects so the effect runs after both signals have been updated.
+
+## Efficient data management
+
+### Memo
+
+Memo, short for memoization, is a technique that is used in reactive systems to enhance performance. 
+
+In the context of reactivity, memos become valuable with their ability to selectively recalculate based on changes in specific signals. 
+This means that a computed value or function can intelligently determine when to recalculate based on the signals it depends on. 
+
+#### Caching
+
+This approach involves the caching of results obtained from resource-intensive function calls. 
+Instead of recalculating the output each time a function is invoked, the cached result is returned when the same inputs are encountered again. 
+
+By remembering the last computed value and the conditions (or signal values) under which it was computed, the system can avoid unnecessary recalculations.
+
+#### Dependency tracking
+
+An essential aspect of effective memoization in a reactive system is accurate dependency tracking.  
+This involves keeping track of the signals upon which a computed value relies. 
+This way, the system can identify when the cached value needs to be invalidated and recomputed. 
+
+This fine-grained dependency tracking ensures that the system's reactivity remains reliable and accurate.
+
+### Store
+
+Stores are important as they provide a structured and centralized approach to state management. 
+
+A store in reactive programming is essentially a container for state. 
+Unlike dealing with individual signals, stores offer a more organized way to manage state, especially in complex applications.
+
+#### State management
+
+At their core, stores serve as containers for state. 
+They facilitate centralized management of state by offering methods to update, reset, or manipulate it. 
```

Comment:
I don't think stores offer any method for resetting anything, like resetting implies that the original value is stored somewhere, but stores don't keep track of that kind of thing, so they can't provide a reset method.